core/src/main/scala/org/apache/spark/ui/jobs/StageTable.scala

```diff
@@ -284,18 +310,52 @@ private[ui] class StagePagedTable(
           {UIUtils.makeProgressBar(started = stageData.numActiveTasks,
           completed = stageData.numCompleteTasks, failed = stageData.numFailedTasks,
           skipped = 0, reasonToNumKilled = stageData.killedTasksSummary, total = info.numTasks)}
-        </td>
-        <td>{data.inputReadWithUnit}</td>
-        <td>{data.outputWriteWithUnit}</td>
-        <td>{data.shuffleReadWithUnit}</td>
-        <td>{data.shuffleWriteWithUnit}</td> ++
-        {
-          if (isFailedStage) {
-            failureReasonHtml(info)
-          } else {
-            Seq.empty
-          }
-        }
+        </td> ++
+        {if (metrics.contains("Input")) {
```

Comment:
It could be interesting to extract all this string to constants to avoid to look for all places if the string changes.
Don't you think?

Reply:
good in general but I think out-of-scope for this PR